### PR TITLE
[DRAFT] [Disaggregated Coordinator] Coordinator wait for resource managers 

### PR DESCRIPTION
**Pull request by swapsmagic**
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java [hunk 1]
```diff
@@ -94,7 +98,13 @@ public LocalDispatchQuery(
     public void startWaitingForResources()
     {
         if (stateMachine.transitionToWaitingForResources()) {
-            waitForMinimumWorkers();
+            if(isResourceManagerEnabled)
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java:101]: I think the pattern here should be more along the line of chaining futures.  When we wait for the resource manager, we should also ensure the minimum coordinator count is met as well.

Comment (swapsmagic) [reply presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java:101]: the waitForMinimumResourceManager method does the same, once it finds enough resource manager, it triggers waitForMimimumWorkers method and wait for that.

File: presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQueryFactory.java [hunk 2]
```diff
@@ -67,7 +69,8 @@ public LocalDispatchQueryFactory(
             LocationFactory locationFactory,
             Map<Class<? extends Statement>, QueryExecutionFactory<?>> executionFactories,
             ClusterSizeMonitor clusterSizeMonitor,
-            DispatchExecutor dispatchExecutor)
+            DispatchExecutor dispatchExecutor,
```
Comment (tdcmeehan) [inline presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQueryFactory.java:72]: Nit: can we change the ordering to match the field assignment?  I.e. `clusterSizeMonitor` -> `serverConfig` -> `dispatchExecutor`

File: presto-main/pom.xml [hunk 3]
```diff
@@ -226,11 +226,6 @@
             <artifactId>drift-codec-utils</artifactId>
         </dependency>
 
-        <dependency>
```
Comment (tdcmeehan) [inline presto-main/pom.xml]: What is the nature of this change?

**Review by tdcmeehan — APPROVED** [review]

**Pull request closed without merging**